### PR TITLE
Upgrade from Laravel 4.2 to Laravel 5.

### PR DESCRIPTION
**Pull request by DFurnes**
# Changes
Updates Voting App to use Laravel 5. Includes a ton of re-organization to fit new L5 project structure. See the [Upgrade Guide](http://laravel.com/docs/5.0/upgrade#upgrade-5.0) for an outline of major changes. Also removes old Gulp configuration in favor of using [Laravel Elixir](http://laravel.com/docs/5.0/elixir), and removes Codeception tests since they become out of date during the pre-launch crunch time before CGG.

Validation, previously done with the [Laracasts Validation](https://github.com/laracasts/Validation) package, still needs to be refactored to use the new L5 [Controller Validation](http://laravel.com/docs/5.0/validation#controller-validation) or [Form Requests](http://laravel.com/docs/5.0/validation#form-request-validation) objects.

There are probably still some things that are broken in here, but basic functionality all seems sound so I think this is good to merge since it's a whopper of a pull request.

For review: @angaither

### CODE REVIEW

**Comment by DFurnes** [comment]
Doing a little digging to see what's causing build error.

**Comment by DFurnes** [comment]
Fixed and rebased. Needed to run a `composer self-update` before the install step.

File: app/Console/Commands/Inspire.php [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	 */
+	protected $name = 'inspire';
+
+	/**
+	 * The console command description.
+	 *
+	 * @var string
+	 */
+	protected $description = 'Display an inspiring quote';
+
```
Comment (angaither) [inline app/Console/Commands/Inspire.php]: lol y?

File: .gitignore [hunk 2]
```diff
@@ -1,20 +1,8 @@
...
-bower_components
-node_modules
-
-/bootstrap/compiled.php
 /vendor
-composer.phar
-composer.lock
-.env.*.php
-.env.php
-.DS_Store
```
Comment (angaither) [inline .gitignore]: leave this bad boy in

File: .gitignore [hunk 3]
```diff
@@ -1,20 +1,8 @@
...
-Thumbs.db
```
Comment (angaither) [inline .gitignore]: this guy toooo

File: app/Http/Middleware/Authenticate.php [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+	 * Handle an incoming request.
+	 *
+	 * @param  \Illuminate\Http\Request  $request
+	 * @param  \Closure  $next
+	 * @return mixed
+	 */
+	public function handle($request, Closure $next)
+	{
+		if ($this->auth->guest())
+		{
```
Comment (angaither) [inline app/Http/Middleware/Authenticate.php]: ew, 
```
if {
 then
}
```

File: app/Console/Commands/Inspire.php (hunk 1 above)
Comment (DFurnes) [inline app/Console/Commands/Inspire.php]: Because we're PHP artisans!

File: app/Http/Middleware/Authenticate.php (hunk 4 above)
Comment (DFurnes) [inline app/Http/Middleware/Authenticate.php]: Yeah, this is default Laravel code. I'm happy to change if we think its worth it.

**Comment by angaither** [comment]
holy PR batman.

I would just say make sure those filtlers are still on the routes, or at least that people can't still hit those guys.

but bueno :+1:

**Comment by DFurnes** [comment]
Updated indentation to spaces, removed default "Inspire" command, and rebased.

**Comment by DFurnes** [comment]
Merging! Definitely will have bug fixes and further testing over the next few PRs.

**Pull request merged**